### DISCUSSION TITLE
Supplementary ebooks not showing up

### DISCUSSION THREAD

**Discussion by delta0815** [advplyr/audiobookshelf#3642] [Q&A] [answered]
Hey all,
I am pretty sure I am doing something wrong but after going through the docs and this community, I decided to ask for help: I have an Audiobook library with a bunch of m4b files (all in one directory). A few of them have supplementary ebooks (pdf) which have the exact same name. When I first created a library in ABS I didn't set the "Audiobooks only" flag, so next to each Audiobook the matching ebook appeared as seperate file. However I would like to have this as a supplement to the Audiobook. However when I enable "Audiobooks only" the PDFs are not being shown anymore (they now appear under error as "missing" and the Audiobook still doesn't show any supplementary files.

What am I doing wrong?

Btw: I created a new library and set the "Audiobooks only" directly from the beginning. Now the PDFs are not shown at all. Neither as missing, nor as a supplement under the details of the Audiobook.

I am stuck.

Thanks for your help!

Cheers,

quicky

**Comment by nichwall** [comment, marked as answer]
If your files are in the root folder (no subfolders) every file will be its own library item. Otherwise, each book must be in its own folder. You are not able to join items within ABS, so you will need to follow the directory structure if you want multiple files associated with the same library item.

https://www.audiobookshelf.org/docs#book-directory-structure

**Comment by delta0815** [comment]
Understood! Thanks for the answer. As I have a process running which automatically monitors my Audible purchases and copies them to one folder to which ABS has access to, I guess I am out of luck and can't use the ABS feature right now. ABS is still the best app I have running on my server. I love it... it's a pity that ABS doesn't feature a logic which groups identical filenames (with different extensions) into one item.

**Reply by nichwall** [reply]
There are a few feature requests about changing the logic to group things like you say (or support various more directory structures), but this requires ground-up rewrite of the backend and database to support so it will likely be a while before it is supported. If you change your script to create a folder per book and then put the files in that folder, then you will be matching the ABS directory structure for multi-file items, you do not need to create the full `Author/Series/Book/files` structure. Libation and OpenAudible support customizing the directory structure they output to as well.
